src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,36 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 3;
+        result = result * 31 + ((width == null) ? 0 : width.hashCode());
+        result = result * 31 + ((length == null) ? 0 : length.hashCode());
+        result = result * 31 + ((color == null) ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+
+        if (width != null ? !width.equals(rectangle.width) : rectangle.width != null) {
+            return false;
+        }
+        if (length != null ? !length.equals(rectangle.length) : rectangle.length != null) {
+            return false;
+        }
+        if (color != null ? !color.equals(rectangle.color) : rectangle.color != null) {
+            return false;
+        }
```

Comment:
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;)

Reply:
please try to improve your previous solution